src/bindings/c/include/openvino/c/ov_partial_shape.h

```diff
@@ -27,48 +42,90 @@ typedef struct ov_partial_shape ov_partial_shape_t;
  */
 
 /**
- * @brief Create a partial shape and initialze with rank and dimension.
+ * @brief Initialze a partial shape with static rank and dynamic dimension.
+ * @ingroup partial_shape
+ * @param rank support static rank.
+ * @param dims support dynamic and static dimension.
+ *  Static rank, but dynamic dimensions on some or all axes.
+ *     Examples: `{1,2,?,4}` or `{?,?,?}` or `{1,2,-1,4}`
+ *  Static rank, and static dimensions on all axes.
+ *     Examples: `{1,2,3,4}` or `{6}` or `{}`
+ *
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_partial_shape_create(ov_partial_shape_t* partial_shape_obj, int64_t rank, ov_dimension_t* dims);
```

Comment:
In the API we have at least two different approaches to function signatures:

- The changed arguments is passed as the first parameter:
  `ov_partial_shape_create(ov_partial_shape_t* partial_shape_obj, int64_t rank, ov_dimension_t* dims);`
- The changed arguments is passed as the last parameter:
  `ov_node_list_get_shape_by_index(const ov_output_node_list_t* nodes, size_t idx, ov_shape_t* shape);`

I understand the difference of these cases but do we have a strict rule which order and when we pass arguments?

Reply:
ok, let add a rule into the document:

 `function parameter order:  input parameters first and then output parameters`

Reply:
We need to change the order of arguments in this method if output parameters are always after input arguments.